### PR TITLE
Vinhlt3/ch134/improve generate unique code of company location

### PR DESCRIPTION
**Pull request by jabraok**

### CODE REVIEW

File: app/models/company.rb [hunk 1]
```diff
@@ -54,23 +54,19 @@ def short_code
 
   private
   def pre_process_location_code_prefix
-    generate_prefix unless has_location_code_prefix?
+    generate_prefix unless valid_prefix?
     self.location_code_prefix = self.location_code_prefix.downcase
```
Comment (brancusi) [inline app/models/company.rb]: Let's remove the `self` from `self.location_code_prefix.downcase`

File: app/models/location.rb [hunk 2]
```diff
@@ -35,24 +35,21 @@ def full_name
 
 	private
 	def pre_process_code
-		generate_code unless has_code?
+		generate_code unless valid_code?
 		self.code = self.code.downcase
```
Comment (brancusi) [inline app/models/location.rb]: Let's remove `self` from `self.code.downcase`

File: app/models/order.rb [hunk 3]
```diff
@@ -133,20 +133,26 @@ def due_date
 
   private
   def setup_defaults
-    generate_order_number
-    set_shipping
+    generate_order_number unless valid_order_number?
+    self.order_number = self.order_number.downcase
```
Comment (brancusi) [inline app/models/order.rb]: Remove `self` from `self.order_number.downcase`

**Comment by brancusi** [comment]
Just a few tweaks, everything looks good

**Comment by jabraok** [comment]
Sorry about those cases. Please check again

**Pull request merged**